packages/core/src/checkbox/CheckboxIcon.tsx

```diff
@@ -68,7 +68,7 @@ export const CheckboxIcon = ({
       className={clsx(
         withBaseName(),
         {
-          [withBaseName("checked")]: checked,
+          [withBaseName("checked")]: checked && !indeterminate,
```

Comment:
I don't think this fix is correct, this would make it harder to apply a different style if you wanted to. Can we fix this by adjusting the CSS specificity please

Reply:
yes, it was my first thought to use CSS but i wasn't sure on the standard for this type of specificity, let me know if there's something preferred than :not selector